ibm/resource_ibm_container_cluster.go

```diff
@@ -952,6 +963,47 @@ func resourceIBMContainerClusterUpdate(d *schema.ResourceData, meta interface{})
 		}
 	}
 
+	if d.HasChange("labels") || d.IsNewResource() {
+		workerPoolsAPI := csClient.WorkerPools()
+		workerPools, err := workerPoolsAPI.ListWorkerPools(clusterID, targetEnv)
+		if err != nil {
+			return err
+		}
+		var poolName string
+		var poolContains bool
+
+		if len(workerPools) > 0 && workerPoolContains(workerPools, defaultWorkerPool) {
+			poolName = defaultWorkerPool
+			poolContains = true
+		} else if len(workerPools) > 0 && workerPoolContains(workerPools, computeWorkerPool) && workerPoolContains(workerPools, gatewayWorkerpool) {
```

Comment:
Why does it also require the `gatewayWorkerpool`?

Reply:
This is following the pattern of the other attribute such as https://github.com/IBM-Cloud/terraform-provider-ibm/blob/78b7ac269954ade9e9ea386c0b643bf15212f2f0/ibm/resource_ibm_container_cluster.go#L940

Reply:
@hkantare can you confirm that the labels are on the gatewayWorkerpool as well?

Reply:
Yes when I checked form UI we can have lables on gateway worker pool also